test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -198,108 +211,185 @@ public Task<HostConcurrencySnapshot> ReadAsync(CancellationToken cancellationTok
                 });
             }
         }
+    }
 
-        internal abstract class TestExtensionScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    public class TestExtensionAStartup : IWebJobsStartup2
+    {
+        public void Configure(IWebJobsBuilder builder)
         {
-            private IOptions<ScaleOptions> _scaleOptions;
-            private IScaleMonitor _scaleMonitor;
-            private ITargetScaler _targetScaler;
-            private TriggerMetadata _triggerMetadata;
+        }
 
-            public TestExtensionScalerProvider(IConfiguration config, IOptions<ScaleOptions> scaleOptions, TriggerMetadata triggerMetadata)
+        public void Configure(WebJobsBuilderContext context, IWebJobsBuilder builder)
+        {
+            if (context.Properties.ContainsKey("IsAzureWebJobsScaleHost"))
```

Comment:
Still thinking what helper we could add to simplify this Properties check - e.g. an IsScaleHost extension method somewhere